src/MailboxesParser/MailboxesParser.php

```diff
@@ -0,0 +1,245 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: Lukasz
+ * Date: 2017-11-10
+ * Time: 19:49
+ */
+
+namespace Ddeboer\Imap\MailboxesParser;
+
+
+use Ddeboer\Imap\Exception\MailboxesParserException;
+use Ddeboer\Imap\MailboxInterface;
+
+class MailboxesParser
+{
+    /** @var MailboxInterface[] */
+    protected $mailboxes;
+    /** @var ParsedMailbox[] */
+    protected $folders;
+    protected $treeStructure;
+
+    const INBOX = 'inbox';
+    const SENT = 'sent';
+    const DRAFT = 'drafts';
+    const SPAM = 'spam';
+    const TRASH = 'trash';
+    const TEMPLATES = 'templates';
+    const ARCHIVES = 'archives';
+
+    protected $specialFoldersIds = [
+        self::INBOX     => ['inbox',],
+        self::SENT      => ['sent', 'sent messages', 'INBOX.Sent', '[Gmail]/Sent Mail',],
+        self::DRAFT     => ['drafts', 'INBOX.Drafts', '[Gmail]/Drafts',],
+        self::SPAM      => ['spam', 'INBOX.spam', '[Gmail]/spam'],
+        self::TRASH     => ['trash', 'bin', 'INBOX.trash', '[Gmail]/trash'],
+        self::TEMPLATES => ['templates'],
+        self::ARCHIVES  => ['archives',],
+    ];
```

Comment:
ID of special folder vary a lot from different email provider, I don't want to have a static list to keep up to date.

What about letting the user specify it in the constructor, so this library doesn't have to maintain it?

Reply:
I think those are common ones. All other folders are simply listed

Reply:
Unfortunately not, those are not as common as you may think.
I work with a lot of IMAP servers where these special folders have many other aliases: only for the Trash one I have 15 variants 😞 in English, without counting localizations.

I'm still against to maintain this list: let the user specify them.

Also the tests would be much more easy to implement and verify.